limiter_atomic_int64.go

```diff
@@ -66,10 +66,10 @@ func (t *atomicInt64Limiter) Take() time.Time {
 		timeOfNextPermissionIssue := atomic.LoadInt64(&t.state)
 
 		switch {
-		case timeOfNextPermissionIssue == 0:
-			// If this is our first request, then we allow it.
+		case timeOfNextPermissionIssue == 0 || (t.maxSlack == 0 && now-timeOfNextPermissionIssue > int64(t.perRequest)):
+			// if this is our first call or t.maxSlack == 0 we need to shrink issue time to now
 			newTimeOfNextPermissionIssue = now
-		case now-timeOfNextPermissionIssue > int64(t.maxSlack):
+		case t.maxSlack > 0 && now-timeOfNextPermissionIssue > int64(t.maxSlack):
```

Comment:
```suggestion
		now-timeOfNextPermissionIssue > int64(t.maxSlack):
```
I think the maxSlack check here is unnecessary - the previous iff should have covered that.

Reply:
I think there can be a case where `t.maxSlack == 0` is true, but `now-timeOfNextPermissionIssue > int64(t.perRequest)` is false, in that case we will try to evaluate `now-timeOfNextPermissionIssue > int64(t.maxSlack)` and it can be true, so we will end up in a wrong branch. So we need `t.maxSlack > 0` check here to step into `default` branch.